lib/manager/components/DeploymentViewer.js

```diff
@@ -21,258 +39,388 @@ export default class DeploymentViewer extends Component {
-            <Col xs={12}>
-              <Table striped hover fill>
-                <thead>
-                  <tr>
-                    <th className='col-md-4'>{getMessage(messages, 'table.name')}</th>
-                    <th>Version</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.dateRetrieved')}</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.loadStatus')}</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.errorCount')}</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.routeCount')}</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.tripCount')}</th>
-                    <th className='hidden-xs'>{getMessage(messages, 'table.stopTimesCount')}</th>
-                    <th>{getMessage(messages, 'table.validFrom')}</th>
-                    <th>{getMessage(messages, 'table.expires')}</th>
-                    <th />
-                  </tr>
-                </thead>
-                <tbody>
-                  {versions.map((version) => {
-                    return <FeedVersionTableRow
-                      feedSource={version.feedSource}
-                      version={version}
-                      project={deployment.project}
-                      deployment={deployment}
-                      key={version.id}
-                      user={user}
-                      updateVersionForFeedSource={updateVersionForFeedSource}
-                      deleteFeedVersionClicked={deleteFeedVersion} />
-                  })}
-                </tbody>
-              </Table>
-            </Col>
-          </Row>
-        </Panel>
       </div>
     )
   }
 }
 
-class FeedVersionTableRow extends Component {
-  static propTypes = {
-    deleteFeedVersionClicked: PropTypes.func,
-    deployment: PropTypes.object,
-    feedSource: PropTypes.object,
-    project: PropTypes.object,
-    updateVersionForFeedSource: PropTypes.func,
-    user: PropTypes.object,
-    version: PropTypes.object
-  }
+class CustomConfig extends Component {
+  state = {}
 
-  _decrementVersion = () => {
-    const {deployment, feedSource, updateVersionForFeedSource, version} = this.props
-    updateVersionForFeedSource(deployment, feedSource, {id: version.previousVersionId})
+  _toggleCustomConfig = evt => {
+    const {deployment, updateDeployment} = this.props
+    const {name} = evt.target
+    const value = deployment[name]
+      ? null
+      : name === 'customBuildConfig'
+        ? SAMPLE_BUILD_CONFIG
+        : SAMPLE_ROUTER_CONFIG
+    updateDeployment(deployment, {[name]: value})
   }
 
-  _incrementVersion = () => {
-    const {deployment, feedSource, updateVersionForFeedSource, version} = this.props
-    updateVersionForFeedSource(deployment, feedSource, {id: version.nextVersionId})
+  _onChangeConfig = ({target}) =>
+    this.setState({[this.props.name]: target.value})
+
+  _isValidJSON = (str) => {
```

Comment:
All functions that are not dependent on instance variables should be singleton functions.  ie: define this as a function in the base of this file so a new function isn't created for each instance.